libraries/Microsoft.Bot.Builder.Dialogs/SkillDialog.cs

```diff
@@ -197,5 +208,66 @@ private async Task<Activity> SendToSkillAsync(ITurnContext context, Activity act
 
             return eocActivity;
         }
+
+        private async Task<bool> InterceptOAuthCardsAsync(ITurnContext turnContext, Activity activity, string connectionName, CancellationToken cancellationToken)
+        {
+            if (activity?.Attachments != null && !string.IsNullOrWhiteSpace(connectionName))
+            {
+                foreach (var attachment in activity.Attachments.Where(a => a?.ContentType == OAuthCard.ContentType))
```

Comment:
Why a for loop? Are we getting many?

Reply:
We have no restrictions but I think this code can be terser; are we looking for the first OAuth card?

Reply:
If the skill sends multiple OAuthCards, do we manually update each card or hope that the user clicks the first OAuthCard and goes down a happy path?

Reply:
I dont want to assume how many oauth cards are sent (though in practical terms, it's always 1).So we try token exchange for all those cards and we show them only if any token exchange fails